de-lib/src/main/java/org/iplantc/de/tools/client/presenter/ManageToolsViewPresenter.java

```diff
@@ -308,16 +310,28 @@ public HandlerRegistration addSelectionChangedHandler(SelectionChangedEvent.Sele
 
     @Override
     public void onShowToolInfo(ShowToolInfoEvent event) {
-       toolInfoDialogProvider.get(new AsyncCallback<ToolInfoDialog>() {
-           @Override
-           public void onFailure(Throwable throwable) {
+        dcService.getAppsForTool(event.getTool().getId(), new AppsCallback<List<App>>() {
+
+            @Override
+            public void onFailure(Integer statusCode, Throwable exception) {
```

Comment:
So it looks like if the tool's apps can't be loaded, then none of the tool details will display?
Maybe we can display an error message and display tool details with an empty apps list instead?